waitress/static/js/dashboard.js

```diff
@@ -0,0 +1,91 @@
+// DOM elements
+const modalBody = document.getElementById('modal-body');
+const refreshSlackUserBtn = document.getElementById('refresh-slack-btn');
+
+
+const notyConfig = {
+    timeout: 1000,
+    closeWith: ['click'],
+    theme: 'relax'
+};
+
+const SPINNER_HTML = `<section class="d-flex justify-content-center align-items-center w-100">
+<div class="spinner-grow text-dark"></div>
+</section>`;
+
+const REFRESH_HTML = '<i class="fas fa-sync-alt"></i>';
+
+export async function refreshSlackUser() {
```

Comment:
This is a bunch of imperative code, I think there are some basic watchers you can set with mutationObserver https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver